tests/azure_iot_hub_client_ut.c

```diff
@@ -381,6 +475,690 @@ static void testAzureIoTHubClient_SendTelemetry_Success( void ** state )
+
+static void testAzureIoTHubClient_DeviceTwinGet_InvalidArgFailure( void ** state )
+{
+    ( void ) state;
+
+    /* Fail GetDeviceTwin when client is NULL */
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( NULL ),
+                      eAzureIoTHubClientInvalidArgument );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_NotSubscribeFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( &xTestIoTHubClient ),
+                      eAzureIoTHubClientTopicNotSubscribed );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_SendFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Subscribe, eAzureIoTMQTTSuccess );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = azureiotmqttPACKET_TYPE_SUBACK;
+    xDeserializedInfo.usPacketIdentifier = usTestPacketId;
+    assert_int_equal( AzureIoTHubClient_SubscribeDeviceTwin( &xTestIoTHubClient,
+                                                             prvTestDeviceTwin,
+                                                             NULL, ( uint32_t ) -1 ),
+                      eAzureIoTHubClientSuccess );
+
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSendFailed );
+    pucPublishPayload = NULL;
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( &xTestIoTHubClient ),
+                      eAzureIoTHubClientPublishFailed );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Subscribe, eAzureIoTMQTTSuccess );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = azureiotmqttPACKET_TYPE_SUBACK;
+    xDeserializedInfo.usPacketIdentifier = usTestPacketId;
+    assert_int_equal( AzureIoTHubClient_SubscribeDeviceTwin( &xTestIoTHubClient,
+                                                             prvTestDeviceTwin,
+                                                             NULL, ( uint32_t ) -1 ),
+                      eAzureIoTHubClientSuccess );
+
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSuccess );
+    pucPublishPayload = NULL;
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( &xTestIoTHubClient ),
+                      eAzureIoTHubClientSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_ReceiveMessages_Success( void ** state )
```

Comment:
Do we have a test where we get a "rando" topic we're not subscribed to?  If not, I think we should add one.  This shouldn't happen, but I'd like to test that (a) no callbacks are called and (b) we don't crash/hang.